tests/modules/submodules/consensus/test_consensus.py

```diff
@@ -130,6 +132,32 @@ def test_get_blockstamp_for_report_slot_not_finalized(web3, consensus, caplog, s
     assert "Reference slot is not yet finalized" in caplog.messages[-1]
 
 
+@pytest.mark.unit
+def test_get_frame_custom_exception(web3, consensus):
+    bs = ReferenceBlockStampFactory.build()
+
+    consensus_contract = Mock(get_current_frame=Mock(side_effect=ContractCustomError('0x12345678', '0x12345678')))
+    consensus._get_consensus_contract = Mock(return_value=consensus_contract)
+
+    with pytest.raises(ContractCustomError):
+        consensus.get_current_frame(bs)
+
+
+@pytest.mark.unit
+def test_first_frame_is_not_yet_started(web3, consensus, caplog, set_no_account):
+    bs = ReferenceBlockStampFactory.build()
+
+    consensus_contract = Mock(get_current_frame=Mock(side_effect=ContractCustomError('0xcd0883ea', '0xcd0883ea')))
+    consensus._get_consensus_contract = Mock(return_value=consensus_contract)
+    consensus.get_frame_config = Mock(return_value=FrameConfigFactory.build(initial_epoch=5, epochs_per_frame=10))
+    consensus.get_chain_config = Mock(return_value=ChainConfigFactory.build())
+
+    first_frame = consensus.get_current_frame(bs)
+
+    assert first_frame.ref_slot == 5 * 32
+    assert first_frame.report_processing_deadline_slot == (5 + 10) * 32 - 1
+
+
```

Comment:
I would expect `FAR_FUTURE_EPOCH` to be handled separate.

```suggestion
@pytest.mark.unit
def test_initial_epoch_is_set_to_far_future_epoch(web3, consensus, caplog, set_no_account):
    bs = ReferenceBlockStampFactory.build()

    consensus_contract = Mock(get_current_frame=Mock(side_effect=ContractCustomError('0xcd0883ea', '0xcd0883ea')))
    consensus._get_consensus_contract = Mock(return_value=consensus_contract)
    consensus.get_frame_config = Mock(
        return_value=FrameConfigFactory.build(
            initial_epoch=48038396025285290,
            epochs_per_frame=10,
        )
    )
    consensus.get_chain_config = Mock(return_value=ChainConfigFactory.build())

    with pytest.raises(ValueError):
        consensus.get_current_frame(bs)
```